From 2683d1414114495e95069da1797bdd34e4bb2236 Mon Sep 17 00:00:00 2001
From: Osamu Aoki <osamu@debian.org>
Date: Fri, 14 Jul 2017 22:05:17 +0900
Subject: [PATCH 4/5] Experimental ssh+sendmail support

This enables to send *@debian.org address mail from
{people,alioth}.debian.org to make its delivery more robust.

This needs some more testing and documentation
---
 debian/debconf/conf.d/router/200_exim4-config_primary    | 10 ++++++++++
 debian/debconf/conf.d/transport/30_exim4-config_ssh_pipe | 15 +++++++++++++++
 debian/exim4-config.install                              |  1 +
 debian/ssh_hosts                                         |  7 +++++++
 4 files changed, 33 insertions(+)
 create mode 100644 debian/debconf/conf.d/transport/30_exim4-config_ssh_pipe
 create mode 100644 debian/ssh_hosts

diff --git a/debian/debconf/conf.d/router/200_exim4-config_primary b/debian/debconf/conf.d/router/200_exim4-config_primary
index 8fee21c..b50901e 100644
--- a/debian/debconf/conf.d/router/200_exim4-config_primary
+++ b/debian/debconf/conf.d/router/200_exim4-config_primary
@@ -73,6 +73,16 @@ nonlocal:
 # domains, you'll need to copy the dnslookup_relay_to_domains router
 # here so that mail to relay_domains is handled separately.
 
+ssh:
+  debug_print = "R: ssh for $local_part@$domain from ${address:$h_from:}"
+  driver = accept
+  condition = ${if exists{CONFDIR/ssh_hosts}\
+      {${lookup{${address:$h_from:}}lsearch{CONFDIR/ssh_hosts}{$value}fail}}fail}
+  domains = *
+  transport = ssh_pipe
+  no_verify
+  no_expn
+
 SUBMISSIONHOST = ${listextract{3}{${lookup{${address:$h_from:}}lsearch{CONFDIR/passwd.client}{$value}fail}}{$value}fail}
 
 submission:
diff --git a/debian/debconf/conf.d/transport/30_exim4-config_ssh_pipe b/debian/debconf/conf.d/transport/30_exim4-config_ssh_pipe
new file mode 100644
index 0000000..47e754b
--- /dev/null
+++ b/debian/debconf/conf.d/transport/30_exim4-config_ssh_pipe
@@ -0,0 +1,15 @@
+
+### transport/30_exim4-config_ssh_pipe
+#################################
+
+SSH_LINE = ${lookup{${address:$h_from:}}lsearch{CONFDIR/ssh_hosts}}
+
+ssh_pipe:
+  debug_print = "T: ssh_pipe for $local_part@$domain from ${address:$h_from:} via SSH_LINE"
+  driver = pipe
+  path = "/bin:/usr/bin:/usr/local/bin"
+  command = "SSH_LINE"
+  return_path_add
+  delivery_date_add
+  envelope_to_add
+
diff --git a/debian/exim4-config.install b/debian/exim4-config.install
index 9ad2aa1..afa367b 100644
--- a/debian/exim4-config.install
+++ b/debian/exim4-config.install
@@ -1,3 +1,4 @@
 debian/debconf/update-exim4.conf.template usr/sbin
 debian/debconf/exim4.conf.template etc/exim4
+debian/ssh_hosts etc/exim4
 debian/script usr/share/bug/exim4-config
diff --git a/debian/ssh_hosts b/debian/ssh_hosts
new file mode 100644
index 0000000..0a120b1
--- /dev/null
+++ b/debian/ssh_hosts
@@ -0,0 +1,7 @@
+# configuration file used when the local exim is
+# authenticating to the remote SSH service host
+# with the account corresponding to the From address as a client.
+# (smarthost,sattelite)
+#
+# Example:
+### header-From:  /usr/bin/ssh -p 22 -i /path/to/ssh-id_rsa foo@example.org /usr/bin/sendmail -bm -ti
-- 
2.11.0

